lib/block_view.go

```diff
@@ -7620,6 +7911,166 @@ func (bav *UtxoView) _connectSwapIdentity(
 	return totalInput, totalOutput, utxoOpsForTxn, nil
 }
 
+// _verifyAccessSignature verifies if the accessSignature is correct. Valid
+// accessSignature is the signed hash of (derivedPublicKey + expirationBlock)
+// in DER format, made with the ownerPublicKey.
+func _verifyAccessSignature(ownerPublicKey []byte, derivedPublicKey []byte,
+	expirationBlock uint64, accessSignature []byte) error {
+
+	// Compute a hash of derivedPublicKey+expirationBlock.
+	expirationBlockBytes := UintToBuf(expirationBlock)
+	accessBytes := append(derivedPublicKey, expirationBlockBytes[:]...)
+	accessHash := Sha256DoubleHash(accessBytes)
+
+	// Convert ownerPublicKey to *btcec.PublicKey.
+	if len(ownerPublicKey) != btcec.PubKeyBytesLenCompressed {
+		fmt.Errorf("_verifyAccessSignature: Problem parsing owner public key")
+	}
+	ownerPk, err := btcec.ParsePubKey(ownerPublicKey, btcec.S256())
+	if err != nil {
+		return errors.Wrapf(err, "_verifyAccessSignature: Problem parsing owner public key: ")
+	}
+
+	// Convert accessSignature to *btcec.Signature.
+	signature, err := btcec.ParseDERSignature(accessSignature, btcec.S256())
+	if err != nil {
+		return errors.Wrapf(err, "_verifyAccessSignature: Problem parsing access signature: ")
+	}
+
+	// Verify signature.
+	if !signature.Verify(accessHash[:], ownerPk) {
+		return fmt.Errorf("_verifyAccessSignature: Invalid signature")
+	}
+
+	return nil
+}
+
+func (bav *UtxoView) _connectAuthorizeDerivedKey(
+	txn *MsgBitCloutTxn, txHash *BlockHash, blockHeight uint32, verifySignatures bool) (
+	_totalInput uint64, _totalOutput uint64, _utxoOps []*UtxoOperation, _err error) {
+
```

Comment:
You need to do one more thing here, which is you need to gate this function on a block height to avoid causing a fork. We have a block height that is triggering Wednesday called NFTTransferOrBurnBlockHeight. You need to do the following:

1. Rename NFTTransferOrBurnBlockHeight to NFTTransferOrBurnAndDerivedKeysBlockHeight
2. Add a block like this to the beginning of this function: https://github.com/bitclout/core/blob/main/lib/block_view.go#L7192
3. Fix your tests by setting the value to zero like line 19225 here: https://github.com/bitclout/core/blob/main/lib/block_view_test.go

Reply:
Yes indeed, we don't want a fork! Thanks for the detailed steps :)